src/terminal/Grid.h

```diff
@@ -147,6 +147,201 @@ struct LogicalLine
+                if (result != 0)
+                {
+                    // Match the remaining text
+                    std::u32string_view remainingTextToMatch(searchText.data() + result,
+                                                             searchText.size() - result);
+                    if (matchTextAt(remainingTextToMatch, ColumnOffset(0), line + 1))
+                        return CellLocation { i, ColumnOffset(static_cast<int>(lineLength - result)) };
+                }
+                startPosition = ColumnOffset(0);
+                ++i;
+            }
+            return std::nullopt;
+        }
+        for (auto line = lines.begin(); line != lines.end(); ++line)
+        {
+            auto result = line->get().search(searchText, startPosition);
+            if (result.has_value())
+            {
+                if (result->partialMatchLength == 0)
+                    return CellLocation { i, result->column };
+                auto remainingText = searchText;
+                remainingText.remove_prefix(result->partialMatchLength);
+                if (line + 1 != lines.end() && (line + 1)->get().matchTextAt(remainingText, ColumnOffset(0)))
+                    return CellLocation { i,
+                                          ColumnOffset::cast_from(
+                                              static_cast<int>(unbox<size_t>(line->get().size())
+                                                               - result->partialMatchLength)) };
+            }
+            startPosition = ColumnOffset(0);
+            ++i;
+        }
+        return std::nullopt;
+    }
+
+    // Searches from right to left, taking into account line wrapping
+    [[nodiscard]] std::optional<terminal::CellLocation> searchReverse(std::u32string_view searchText,
+                                                                      ColumnOffset startPosition) const
+    {
+        auto i = bottom;
+        auto const lineLength = unbox<size_t>(lines.front().get().size());
+        if (searchText.size() > lineLength)
+        {
+            for (auto line = lines.rbegin(); line != lines.rend(); ++line)
+            {
+                std::u32string_view textOnThisLine(searchText.data() + searchText.size()
+                                                       - unbox<size_t>(startPosition),
+                                                   unbox<size_t>(startPosition));
+                auto const result = searchPartialMatchReverse(textOnThisLine, line->get());
+                if (result != 0)
+                {
+                    std::u32string_view remainingText(searchText.data(), searchText.size() - result);
+                    // Check if the searchText can even fit in the available lines
+                    auto const willFit = [&] {
+                        auto const count =
+                            static_cast<size_t>(std::max(0l, std::distance(line + 1, lines.rend())));
```

Comment:
In this line, MSVC/Windows CI build is failing. It cannot find a proper overload for `std::max(A, B)` most likely because the type of A and B is not equal, which it must be.

```suggestion
                            static_cast<size_t>(std::max(long { 0 }, static_cast<long>(std::distance(line + 1, lines.rend()))));
```

This could do it. I'm sorry, I didn't verify, as I'm just typing in my web browser, but you simply need to make sure the type of each parameter is equal to each other. :)